### PR TITLE
ACMS-3281: Refactor Drupal recommended settings and add tests.

### PR DESCRIPTION
**Pull request by rajeshreeputra**
**Motivation**
Fixes #[ACMS-3281](https://acquia.atlassian.net/browse/ACMS-3281)

**Proposed changes**

**Alternatives considered**

**Testing steps**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -29,22 +25,15 @@
         "loophp/phposinfo": "^1.7.1",
         "symfony/yaml": "^4.4 || ^5 || ^6"
     },
+    "conflict": {
+        "acquia/blt-multisite": "*"
+    },
```
Comment (chandan-singh7929) [inline composer.json:30]: Don't add this conflict as of now, will check first if needed will be added as part of multisite implementation ticket

File: README.md [hunk 2]
```diff
@@ -1,14 +1,15 @@
 # Acquia Drupal Recommended Settings
-The Acquia Drupal Recommended Settings plugin adds the recommended settings to the Drupal project, so developers won't have to edit settings.php manually.
+The Acquia Drupal Recommended Settings plugin adds the recommended settings to
+the Drupal project, so developers won't have to edit settings.php manually.
 
 The recommended settings includes:
-- the required database credentials
+- the required database credentials.
```
Comment (apathak18) [inline README.md:6]: If these are pointers then first word of each should be in caps.
- The required database credentials.
- Configuration sync directory path.
-  File directory path i.e public/private etc.
- Acquia site studio sync directory path.

File: src/Settings.php [hunk 3]
```diff
@@ -136,7 +136,6 @@ public function generate(array $overrideData = []): void {
 
     // The config directory for given site must exists, otherwise Drupal will
     // add database credentials to settings.php.
-    // @todo Using hardcoded sync directory path. Need to better optimise it.
```
Comment (apathak18) [inline src/Settings.php:139]: is this done?

File: README.md (hunk 2 above)
Comment (rajeshreeputra) [reply README.md]: done

**Review by apathak18 — APPROVED** [review]

File: tests/unit/SettingsTest.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+   */
+  public function setUp(): void {
+    $this->drupalRoot = dirname(__FILE__);
+    $docroot = $this->drupalRoot . '/docroot';
+    $this->drsFileSystem = new DrsFilesystem();
+    $this->drsFileSystem->ensureDirectoryExists($docroot . '/sites/default');
+    $this->fileSystem = new Filesystem();
+    $this->fileSystem->touch($docroot . '/sites/default/default.settings.php');
+    $this->settings = new Settings($docroot, "default");
+    $this->settings->generate();
```
Comment (chandan-singh7929) [inline tests/unit/SettingsTest.php:51]: Pass db_spec here like
`$this->settings->generate();`

File: tests/unit/SettingsTest.php [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+ * @link https://docs.acquia.com/
+ */
+';
+    $this->assertEquals($content, file_get_contents($this->drupalRoot . '/docroot/sites/default/settings.php'));
+    // Assert that default.includes.settings.php file exist.
+    $this->assertTrue($this->fileSystem->exists($this->drupalRoot . '/docroot/sites/default/settings/default.includes.settings.php'));
+    // Assert that default.local.settings.php file exist.
+    $this->assertTrue($this->fileSystem->exists($this->drupalRoot . '/docroot/sites/default/settings/default.local.settings.php'));
+    // Assert that local.settings.php file exist.
+    $this->assertTrue($this->fileSystem->exists($this->drupalRoot . '/docroot/sites/default/settings/local.settings.php'));
```
Comment (chandan-singh7929) [inline tests/unit/SettingsTest.php:80]: And then check that db_specs are replaced correctly in local.settings.php.

**Review by chandan-singh7929 — CHANGES_REQUESTED** [review]
Overall all changes looks good to me, there's minor suggestion please check

File: tests/unit/SettingsTest.php (hunk 4 above)
Comment (rajeshreeputra) [reply tests/unit/SettingsTest.php]: Incorporated.

File: tests/unit/SettingsTest.php (hunk 5 above)
Comment (rajeshreeputra) [reply tests/unit/SettingsTest.php:90]: Incorporated

**Pull request merged**